healthcheck/healthcheck.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package healthcheck provides simple health check implementations.
+package healthcheck
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+)
+
+type HTTPResponse struct {
+	Message string
+}
+
+// A basic http healthcheck implementation.
+func HandleHTTPHealth() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+
+		contentType := r.Header.Get("Content-Type")
+		w.Header().Set("Content-Type", contentType)
+
+		switch contentType {
+		case "application/json":
```

Comment:
This won't actually work 😦. Chrome sends `application/json; charset=utf-8`. You can either use [ParseMediaType](https://pkg.go.dev/mime#ParseMediaType) (which is pretty expensive) or check "HasPrefix" instead.